### PR TITLE
fix: bsc savers user staking data

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Makes sure BSC savers references are added everywhere, so that BSC user staking data is properly fetched - it accidentally works on deposit/withdraw currently, as we use `<TransactionProvider />` user-data fetching introspecting THOR Tx, but won't fetch the user staking data when importing a wallet.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/5415

## Risk

None

## Testing

- Have a wallet with a savers BNB active position (can use e.g a diff env for that, or deposit then clear your localForage)
- Savers user staking data for BNB on BSC is fetched on wallet import, without needing to do a deposit

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

- Develop

Screenshot 2023-10-09 at 10 24 04

- This diff

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5435** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5435" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5435?utm_source=stack-comment).

File: src/context/AppProvider/AppContext.tsx [hunk 1]
```diff
@@ -213,6 +214,13 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
           case dogeChainId:
           case bchChainId:
           case cosmosChainId:
+          case bscChainId:
```
Comment (kaladinlight) [inline src/context/AppProvider/AppContext.tsx:217]: fallthrough?

Comment (gomesalexandre) [reply src/context/AppProvider/AppContext.tsx:217]: The `ethChainId` has slightly more logic - if we're fine with turning `no-duplicate-case` off here, might as well!

Comment (kaladinlight) [reply src/context/AppProvider/AppContext.tsx:217]: was just thinking with the avalanche chain id

Comment (gomesalexandre) [reply src/context/AppProvider/AppContext.tsx:217]: Absolutely - 7a101c1691

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**